cmake/adios2-config-common.cmake.in

```diff
@@ -85,6 +87,21 @@ find_package_handle_standard_args(${CMAKE_FIND_PACKAGE_NAME} CONFIG_MODE)
 
 if(NOT TARGET adios2::adios2)
   include("${CMAKE_CURRENT_LIST_DIR}/adios2-targets.cmake")
+  # Filter our usage requirements to avoid non-enabled languages.
+  if(NOT CMAKE_C_COMPILER_LOADED)
+    # Since C is not available, use MPI through MPI_CXX instead.
+    get_property(_adios2_link_libs TARGET adios2::adios2 PROPERTY INTERFACE_LINK_LIBRARIES)
+    string(REGEX REPLACE "MPI::MPI_C(;|>|$)" "MPI::MPI_CXX\\1" _adios2_link_libs "${_adios2_link_libs}")
+    set_property(TARGET adios2::adios2 PROPERTY INTERFACE_LINK_LIBRARIES "${_adios2_link_libs}")
+    unset(_adios2_link_libs)
+  endif()
+  if(NOT CMAKE_CXX_COMPILER_LOADED)
+    # Avoid requiring compile features for non-enabled languages.
+    get_property(_adios2_cxx_features TARGET adios2::adios2 PROPERTY INTERFACE_COMPILE_FEATURES)
+    list(FILTER _adios2_cxx_features EXCLUDE REGEX "^cxx_")
+    set_property(TARGET adios2::adios2 PROPERTY INTERFACE_COMPILE_FEATURES "${_adios2_cxx_features}")
+    unset(_adios2_cxx_features)
```

Comment:
That CMake errors on this case is perhaps incorrect.  I've opened [CMake Issue 20438](https://gitlab.kitware.com/cmake/cmake/issues/20438) for it.